pkg/controller.v1beta1/experiment/manifest/generator_test.go

```diff
@@ -335,8 +335,8 @@ spec:
 		} else if !tc.Err {
 			if err != nil {
 				t.Errorf("Case: %v failed. Expected nil, got %v", tc.testDescription, err)
-			} else if !reflect.DeepEqual(expectedRunSpec, actualRunSpec) {
-				t.Errorf("Case: %v failed. Expected %v\n got %v", tc.testDescription, expectedRunSpec.Object, actualRunSpec.Object)
+			} else if diff := cmp.Diff(expectedRunSpec.Object, actualRunSpec.Object); diff != "" {
+				t.Errorf("Case: %v failed. Diff (-expected, +actual):\n%s", tc.testDescription, diff)
```

Comment:
We must define errors, and compare expected and wanted errors and values.

Define errors: https://github.com/kubeflow/katib/blob/6fb9157a4dd0ecec34ae35233965c30c93918f34/pkg/util/v1beta1/katibconfig/config.go#L38-L42
Compare Errors: https://github.com/kubeflow/katib/blob/6fb9157a4dd0ecec34ae35233965c30c93918f34/pkg/util/v1beta1/katibconfig/config_test.go#L72-L74

Reply:
Sounds good. I am creating a separate PR given that [file-metricscollector_test.go](https://github.com/kubeflow/katib/pull/2274/files#diff-170ce85bf8b4c15b6576e29ceba0077ef91cffa6bf4bd36ac602c6f450406e1e) has already been updated in https://github.com/kubeflow/katib/pull/2274.